include/seqan3/range/decorator/gap_decorator.hpp

```diff
@@ -336,6 +339,163 @@ class gap_decorator
         //!\}
     };
 
+    /*!\brief The iterator that moves over the seqan3::gap_decorator with random access semantics.
+     *
+     * \details
+     *
+     * \attention This iterator models std::RandomAccessIterator but **does not guarantee constant time access**.
+     *            Random access is provided in \f$O(\log k)\f$ time (where `k` is the number of gaps).
+     *
+     * This iterator returns values when dereferenced, not references, i.e. it does not model the
+     * [Cpp17RandomAccessIterator](https://en.cppreference.com/w/cpp/named_req/RandomAccessIterator)
+     * requirements of the STL. It does model the C++20 std::RandomAccessIterator.
+     */
+    class gap_decorator_iterator_ra : public gap_decorator_iterator
+    {
+    private:
+        //!\brief The base type.
+        using base_t = gap_decorator_iterator;
+
+    public:
+        /*!\name Associated types
+         * \brief Make the parent's member types visible.
+         * \{
+         */
+        //!\brief Type for distances between iterators.
+        using difference_type = typename gap_decorator::difference_type;
+        //!\brief Value type of container elements.
+        using value_type = typename gap_decorator::value_type;
+        //!\brief Const-reference type defined by container (which equals the reference type).
+        using reference = typename gap_decorator::const_reference; // = reference
+        //!\brief Equals reference type.
+        using const_reference = reference;
```

Comment:
why do you need to define this?

Reply:
This is required by a container.. since we do not model container I could remove this. I don't care?

Reply:
@rrahn ping

Reply:
But it is not required by the iterator. So please remove it then.